backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java

```diff
@@ -67,7 +71,35 @@ public void getExperimentRunById(
   public void getExperimentRunByName(
       GetExperimentRunByName request,
       StreamObserver<GetExperimentRunByName.Response> responseObserver) {
-    super.getExperimentRunByName(request, responseObserver);
+    try {
+      final var response =
+          futureExperimentRunDAO
+              .findExperimentRuns(
+                  FindExperimentRuns.newBuilder()
+                      .addPredicates(
+                          KeyValueQuery.newBuilder()
+                              .setKey("name")
+                              .setValue(
+                                  Value.newBuilder().setStringValue(request.getName()).build())
+                              .setOperator(OperatorEnum.Operator.EQ)
+                              .setValueType(ValueTypeEnum.ValueType.STRING)
+                              .build())
+                      .build())
+              .thenApply(
+                  findResponse -> {
+                    if (findResponse.getExperimentRunsCount() == 0) {
+                      return GetExperimentRunByName.Response.newBuilder().build();
```

Comment:
Should we return 404 in this case?

Reply:
Added 404 while not found the run by name